src/main/scala/encry/view/wallet/WalletStorage.scala

```diff
@@ -18,24 +21,29 @@ case class WalletStorage(store: Store, publicKeys: Set[PublicKey25519]) extends
   def getBoxById(id: ADKey): Option[EncryBaseBox] = store.get(keyByBoxId(id))
     .flatMap(d => StateModifierDeserializer.parseBytes(d.data, id.head).toOption)
 
-  def allBoxes: Seq[EncryBaseBox] = store.getAll
-    .filter(_._2 != balancesKey)
-    .foldLeft(Seq[EncryBaseBox]()) { case (acc, id) =>
-      getBoxById(ADKey @@ id._1.data).map(bx => acc :+ bx).getOrElse(acc)
-    }
+  def allBoxes: Seq[EncryBaseBox] = {
+    val startTime: Amount = System.nanoTime()
+    val storeGetAllSeq: Iterator[(K, V)] = store.getAll
+    val endStoreGetAllTime: Amount = System.nanoTime() - startTime
+    system.actorSelection("user/statsSender") ! GetAllTiming(endStoreGetAllTime, storeGetAllSeq.size)
+
+    val outputs: Seq[EncryBaseBox] = store.getAll.filter { x => x._2 != balancesKey }.foldLeft(Seq[EncryBaseBox]()) {
+      case (acc, id) => getBoxById(ADKey @@ id._1.data).map { bx => acc :+ bx }.getOrElse(acc) }
```

Comment:
I think you can move filtering inside foldLeft, so, you will make only one iteration over collection